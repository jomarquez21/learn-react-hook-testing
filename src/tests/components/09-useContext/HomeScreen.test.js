import React from 'react';
import {mount, shallow} from 'enzyme';
import {HomeScreen} from './../../../components/09-useContext/HomeScreen';
import {UserContext} from './../../../components/09-useContext/UserContext';

describe('Test on `HomeScreen.js`', () => {
  const user = {
    name: 'Jomarquez',
    email: 'jmrqz21@gmai.com'
  };

  const component = mount(
    <UserContext.Provider value={{user, setUser: jest.fn()}}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test('Debe renderizar correctamente', () => {
    expect(component).toMatchSnapshot();
  });
});