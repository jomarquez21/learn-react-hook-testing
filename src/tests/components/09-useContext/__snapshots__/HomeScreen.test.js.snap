// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on \`HomeScreen.js\` Debe renderizar correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Jomarquez",
   "email": "jmrqz21@gmai.com"
}
  </pre>
</div>
`;
